### PR TITLE
cdc: always ignore sequence tables

### LINKED ISSUE #4552: Sequence appears in TiCDC ineligible table list
### What did you do?

```shell
mysql> CREATE SEQUENCE SEQ_TEST_NO INCREMENT BY 1 START WITH 1 MAXVALUE 999999999999999 NOMINVALUE CYCLE NOCACHE;

$ bin/cdc cli changefeed create \
    --sink-uri="blackhole://" -c "blackhole-seq"
    --no-confirm

[WARN] some tables are not eligible to replicate, []model.TableName{model.TableName{Schema:"test", Table:"SEQ_TEST_NO ", TableID:0, IsPartition:false}}
Create changefeed successfully!
ID: blackhole-seq
Info: {"sink-uri":"blackhole://","opts":{},"create-time":"2022-02-10T11:32:08.692371203+08:00","start-ts":431086351275786296,"target-ts":0,"admin-job-type":0,"sort-engine":"unified","sort-dir":"","config":{"case-sensitive":true,"enable-old-value":true,"force-replicate":false,"check-gc-safe-point":true,"filter":{"rules":["test.*"],"ignore-txn-start-ts":[1,2]},"mounter":{"worker-num":16},"sink":{"dispatchers":null,"protocol":"","column-selectors":null},"cyclic-replication":{"enable":false,"replica-id":0,"filter-replica-ids":null,"id-buckets":0,"sync-ddl":false},"scheduler":{"type":"table-number","polling-time":-1},"consistent":{"level":"none","max-log-size":64,"flush-interval":1000,"storage":""}},"state":"normal","error":null,"sync-point-enabled":false,"sync-point-interval":600000000000,"creator-version":"v5.4.0-alpha"}
```

<details> <summary><b>DDL Job and Table Info</b></summary>

```json
{
    "id": 20536,
    "type": 34,
    "schema_id": 20531,
    "table_id": 20535,
    "schema_name": "test",
    "state": 6,
    "err": null,
    "err_count": 0,
    "row_count": 0,
    "raw_args": null,
    "schema_state": 5,
    "snapshot_ver": 0,
    "real_start_ts": 0,
    "start_ts": 431086310747799612,
    "dependency_id": 0,
    "query": "CREATE SEQUENCE SEQ_TEST_NO INCREMENT BY 1 START WITH 1 MAXVALUE 999999999999999 NOMINVALUE CYCLE NOCACHE",
    "binlog": {
        "SchemaVersion": 10327,
        "DBInfo": null,
        "TableInfo": {
            "id": 20535,
            "name": {
                "O": "SEQ_TEST_NO",
                "L": "seq_test_no"
            },
            "charset": "",
            "collate": "",
            "cols": null,
            "index_info": null,
            "constraint_info": null,
            "fk_info": null,
            "state": 5,
            "pk_is_handle": false,
            "is_common_handle": false,
            "common_handle_version": 0,
            "comment": "",
            "auto_inc_id": 0,
            "auto_id_cache": 0,
            "auto_rand_id": 0,
            "max_col_id": 0,
            "max_idx_id": 0,
            "max_cst_id": 0,
            "update_timestamp": 431086310761168915,
            "ShardRowIDBits": 0,
            "max_shard_row_id_bits": 0,
            "auto_random_bits": 0,
            "pre_split_regions": 0,
            "partition": null,
            "compression": "",
            "view": null,
            "sequence": {
                "sequence_start": 1,
                "sequence_cache": false,
                "sequence_cycle": true,
                "sequence_min_value": 1,
                "sequence_max_value": 999999999999999,
                "sequence_increment": 1,
                "sequence_cache_value": 0,
                "sequence_comment": ""
            },
            "Lock": null,
            "version": 4,
            "tiflash_replica": null,
            "is_columnar": false,
            "temp_table_type": 0,
            "cache_table_status": 0,
            "policy_ref_info": null,
            "stats_options": null
        },
        "FinishedTS": 431086310774276154,
        "MultipleTableInfos": null
    },
    "version": 1,
    "reorg_meta": null,
    "multi_schema_info": null,
    "priority": 0
}
```

```json
{
 "id": 20535,
 "name": {
  "O": "SEQ_TEST_NO",
  "L": "seq_test_no"
 },
 "charset": "",
 "collate": "",
 "cols": null,
 "index_info": null,
 "constraint_info": null,
 "fk_info": null,
 "state": 5,
 "pk_is_handle": false,
 "is_common_handle": false,
 "common_handle_version": 0,
 "comment": "",
 "auto_inc_id": 0,
 "auto_id_cache": 0,
 "auto_rand_id": 0,
 "max_col_id": 0,
 "max_idx_id": 0,
 "max_cst_id": 0,
 "update_timestamp": 431086310761168915,
 "ShardRowIDBits": 0,
 "max_shard_row_id_bits": 0,
 "auto_random_bits": 0,
 "pre_split_regions": 0,
 "partition": null,
 "compression": "",
 "view": null,
 "sequence": {
  "sequence_start": 1,
  "sequence_cache": false,
  "sequence_cycle": true,
  "sequence_min_value": 1,
  "sequence_max_value": 999999999999999,
  "sequence_increment": 1,
  "sequence_cache_value": 0,
  "sequence_comment": ""
 },
 "Lock": null,
 "version": 4,
 "tiflash_replica": null,
 "is_columnar": false,
 "temp_table_type": 0,
 "cache_table_status": 0,
 "policy_ref_info": null,
 "stats_options": null
}
```

</details>

### What did you expect to see?

Base on [TiCDC doc](https://docs.pingcap.com/tidb/stable/ticdc-overview#unsupported-scenarios), we should ignore SEQUENCE DDL, so we should also ignore its data (include table info).

### What did you see instead?

See above.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

TiCDC version (execute `cdc version`):

```console
v5.4.0
```

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #4552 

### What is changed and how it works?

Always ignore sequence tables

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix a bug that sequence should not be replicated even if force-replication is true.
```

### CODE REVIEW

File: cdc/api/validator.go [hunk 1]
```diff
@@ -222,6 +224,11 @@ func verifyTables(replicaConfig *config.ReplicaConfig, storage tidbkv.Storage, s
 		if filter.ShouldIgnoreTable(tableInfo.TableName.Schema, tableInfo.TableName.Table) {
 			continue
 		}
+		// Sequence is not supported yet, TiCDC needs to filter all sequence tables.
+		// See https://github.com/pingcap/tiflow/issues/4559
```
Comment (ben1009) [inline cdc/api/validator.go:228]: the fix is ignore sequence tables no matter mysql or tidb ?

**Comment by ben1009** [comment]
```
[2022-02-10T13:34:22.486Z] === FAIL: pkg/p2p TestClientSendAnomalies (unknown)

10T13:34:22.487Z] assert: mock: I don't know what to return because the method call was unexpected.

[2022-02-10T13:34:22.487Z] 	Either do Mock.On("Flush").Return(...) first, or remove the Flush() call.

[2022-02-10T13:34:22.487Z] 	This method was unexpected:

[2022-02-10T13:34:22.487Z] 		Flush()

[2022-02-10T13:34:22.487Z] 		

[2022-02-10T13:34:22.487Z] 	at: [client_test.go:45 client.go:319 client.go:211 client.go:199 errgroup.go:57 asm_amd64.s:1371]
```

File: cdc/owner/schema.go [hunk 2]
```diff
@@ -116,48 +116,25 @@ func (s *schemaWrap4Owner) BuildDDLEvent(job *timodel.Job) (*model.DDLEvent, err
...
 		// skip the mark table if cyclic is enabled
 		return true
 	}
-	if !tableInfo.IsEligible(s.config.ForceReplicate) {
-		log.Warn("skip ineligible table", zap.Int64("tid", tableInfo.ID), zap.Stringer("table", tableInfo.TableName))
+	if !t.IsEligible(s.config.ForceReplicate) {
+		// Sequence is not supported yet, and always ineligible.
+		// Skip Warn to avoid confusion.
+		// See https://github.com/pingcap/tiflow/issues/4559
+		if t.IsSequence() {
```
Comment (asddongmen) [inline cdc/owner/schema.go:133]: Seems that we already check `t.IsSequence()` in `t.IsEligible()`, why do you check it here again?

**Review by ben1009 — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**